scripts/dashboard/Activity.vue

```diff
@@ -86,6 +87,23 @@ async function update(txToAdd = 0) {
     const orderedTxs = Array.from(wallet.getTransactions()).sort(
         (a, b) => a.blockHeight - b.blockHeight
     );
+
+    // For Rewards: aggregate the total amount
+    if (props.rewards) {
+        for (const tx of orderedTxs) {
+            // Only compute rewards
+            if (!tx.isCoinStake()) continue;
+            // If this Tx Height is over the last scanned height, we aggregate it
+            if (tx.blockHeight > nRewardUpdateHeight) {
```

Comment:
`orderedTxs` is the array of transactions ordered by blockHeight, therefore you don't have to loop over all its elements.
```suggestion
            if(tx.blockHeight <= nRewardUpdateHeight) break;
```

Reply:
Applied